client/repoutils.c

```diff
@@ -458,7 +458,10 @@ TDNFRemoveRpmCache(
     }
     if (!IsNullOrEmptyString(pszRpmCacheDir))
     {
-        rmdir(pszRpmCacheDir);
+        if (rmdir(pszRpmCacheDir))
+        {
+            dwError = errno;
+        }
```

Comment:
I think we should have a `BAIL_ON_TDNF_ERROR` here. I know it makes no difference, but it would be cleaner.

Reply:
The idea here is, not to end up in an infinite loop. If I use BAIL_ON_TDNF_ERROR we go to error: and then again come back to cleanup: which makes it difficult to read the code because of jumps(which is why goto statements are discouraged). I personally prefer code moving in single direction, not back and forth.

Reply:
Yeah, it's a corner case. I'll leave t up to you. Maybe add a comment.

But - I just spotted another issue with this: this would make the function return `errno`. `BAIL_ON_TDNF_SYSTEM_ERROR` would make it return `ERROR_TDNF_SYSTEM_BASE + dwError`. Functions are expected to return TDNF errors, not system errors.

Reply:
Good catch, fixed it. Thanks.